collab-database/src/workspace_database/body.rs

```diff
@@ -85,17 +86,21 @@ impl WorkspaceDatabase {
   }
 
   /// Update the database by the given id
-  pub fn update_database(&mut self, database_id: &str, f: impl FnMut(&mut DatabaseMeta)) {
-    self
-      .body
-      .update_database(&mut self.collab.transact_mut(), database_id, f);
+  pub fn update_database(
+    &mut self,
+    database_id: &str,
+    f: impl FnMut(&mut DatabaseMeta),
+  ) -> TransactionMut {
+    let mut txn = self.collab.transact_mut();
+    self.body.update_database(&mut txn, database_id, f);
+    txn
   }
 
   /// Delete the database by the given id
-  pub fn delete_database(&mut self, database_id: &str) {
-    self
-      .body
-      .delete_database(&mut self.collab.transact_mut(), database_id);
+  pub fn delete_database(&mut self, database_id: &str) -> TransactionMut {
+    let mut txn = self.collab.transact_mut();
+    self.body.delete_database(&mut txn, database_id);
```

Comment:
As you can see here, you don't need to return transaction. You can just call `self.collab.transact_mut` + `self.body.delete_database`. Because both self.collab and self.body are split in every collab type, creating transaction should never be a problem.